src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
 
 	return array_intersect( $filtered_values, $autoload_values );
 }
+
+
+/**
+ * Filters site metadata for multisite installations.
+ *
+ * This function is specific to multisite setups and provides a way to retrieve
+ * network options. It enforces parameters' constraints and suggests using an alternative function.
+ *
+ * @param mixed $current The value to return if no changes are made.
+ * @param int $object_id Site ID.
+ * @param string $meta_key Metadata key.
+ * @param bool $single Whether to retrieve a single value. Must be true.
+ *
+ * @return mixed The network option value for the specified meta key, or false if the function is used incorrectly.
+ */
+function filter_get_site_metadata( $current, $object_id, $meta_key, $single ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'get_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( ! $single ) {
+		_doing_it_wrong( 'get_metadata', __( 'The $single parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
```

Comment:
While it is correct that `! $single` does not work when using with network options, I don't think we should allow the function to pass through when `$single` is `false`. I think we have two options here:
* Either return an empty array because it's not valid.
* Or return an array with just the option value as the sole element.

Here is an example for the 2nd option:
```suggestion
		$option_value = get_network_option( $object_id, $meta_key, false );
		if ( false !== $option_value ) {
			return array( $option_value );
		}
		return array();
```

Reply:
If we do this, it would a breaking change. If for some reason you were using network options to store multiple values in the same key, the other keys that become in accessible for this change. My hope is that the doing it wrong, will tell developers to stop doing this and we can change this in the future.

Reply:
This would only be a breaking change if someone was actually using this to store multiple values - which I doubt is common. Probably the most common usage of the metadata functions for network options is via WP-CLI.

I'm on board with not changing the return value here for now to first only warn developers about this. But the warning should only appear if there are actually multiple values stored. The problem is not that someone is calling this function with `$single = false` - the problem is when there are multiple values stored for what should be a single network option.

So we would need to conditionally trigger a warning via another filter, once the value has been retrieved, and only _if_ it's multiple values.

Even more important would be to flag this in the filter callback for `add_metadata`: If there's already a value stored and you call this function, it should not be allowed to store multiple values.

But here we need basically the same warning, in order to account for data that was already stored before this change.